backend/src/equipment/equipment.controller.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  Body,
+  Controller,
+  Post,
+  Get,
+  Delete,
+  Param,
+  Put,
+} from '@nestjs/common';
+import { ScheduleDTO } from '../schedule/dto/schedule.dto';
+import { EquipmentDTO } from './dto/equipment.dto';
+import { EquipmentService } from './equipment.service';
+
+@Controller(EquipmentController.EQUIPMENT_API_PATH)
+export class EquipmentController {
+  public static readonly EQUIPMENT_API_PATH = '/equipment';
+  public static readonly EQUIPMENT_ID_ROUTE = '/:id';
+  public static readonly EQUIPMENT_CONTROLLER_WITH_SCHED = '/schedule/:id';
+  constructor(private readonly equipmentService: EquipmentService) {}
+
+  @Get()
+  async getAllEquipments(): Promise<EquipmentDTO> {
+    return this.equipmentService.getAllEquipments();
+  }
+  @Post()
+  async addEquipment(@Body() data: EquipmentDTO): Promise<EquipmentDTO> {
+    return this.equipmentService.addEquipment(data);
+  }
+  @Post(EquipmentController.EQUIPMENT_CONTROLLER_WITH_SCHED)
+  async addScheduletoEquipment(
+    @Param('id') id: string,
+    @Body() data: ScheduleDTO
+  ): Promise<EquipmentDTO> {
+    return this.equipmentService.addScheduletoEquipment(id, data);
+  }
```

Comment:
it would've been nice if we could delete this method as well and handle adding the schedule to equipment in the `updateEquipment` endpoint but i haven't found a way to do that yet with Prisma. we can leave this for now and i will try to find a way to do it later